### PR TITLE
Netsuite: Use new SDF version, refactor client config, expose a new config option to limit import chunks in SDF

### PR DESCRIPTION
**Pull request by omrilit**
### Background:
SDF has released a new version that supposes to fix the intermittent NPE we receive when fetching types with multiple data.
It was done in SDF JS code by converting the 'ALL' parameter to a list of explicit IDs and chunking them when sending the requests to the java sdk.
The chunk size that was set by SDF was changed from 35 to 8 due to instability. After running several tests I found that setting it to 30 seems stable.
I rebased our fork on top of that fix, and added a new configurable `maxItemsInImportObjectsRequest` argument so we'll be able to control it from salto.

### In this PR you can find:
- using the new SDF version
- enabling to control the configurable `maxItemsInImportObjectsRequest`
- refactor the client config to use an object that will be passed directly to the client
- update documentation accordingly

---
_Release Notes_
Netsuite - Extend configuration options for limiting the number of elements that will be fetched in each importObjects API call, in order to address `Failed to import custom objects` errors for types with plenty of instances .

_Upgrade Instructions_
The `fetchAllTypesAtOnce`, `fetchTypeTimeoutInMinutes`, `sdfConcurrencyLimit` configuration parameters are no longer used. If they were modified from the original value, add the following section under `netsuite.nacl` (with the value from your configuration):
```
  client = {
    sdfConcurrencyLimit = <value>
    fetchTypeTimeoutInMinutes = <value>
    fetchAllTypesAtOnce = <value>
  }
```

### CODE REVIEW

**Comment by gilhoffer** [comment]
@omrilit , when you tagged as `non-backward-compatible`, what did you mean exactly? That the adapter configuration requires a manual transformation, or anything else?

**Comment by omrilit** [comment]
Just to emphasize that a manual upgrade in certain WSs is required as explained above.
Maybe we should have additional labels for `releaseNotes` & `manualUpgrade`?

File: packages/netsuite-adapter/src/config.ts [hunk 1]
```diff
@@ -118,6 +146,18 @@ export const getConfigFromConfigChanges = (failedToFetchAllAtOnce: boolean, fail
...
+  const clientConfigSuggestion = suggestions[FETCH_ALL_TYPES_AT_ONCE] !== undefined
+    ? _.pickBy({
+      [FETCH_ALL_TYPES_AT_ONCE]: suggestions[FETCH_ALL_TYPES_AT_ONCE],
+      [SDF_CONCURRENCY_LIMIT]: currentConfig[CLIENT_CONFIG]?.[SDF_CONCURRENCY_LIMIT],
+      [MAX_ITEMS_IN_IMPORT_OBJECTS_REQUEST]: currentConfig[CLIENT_CONFIG]
+        ?.[MAX_ITEMS_IN_IMPORT_OBJECTS_REQUEST],
+      [FETCH_TYPE_TIMEOUT_IN_MINUTES]: currentConfig[CLIENT_CONFIG]
+        ?.[FETCH_TYPE_TIMEOUT_IN_MINUTES],
+    }, values.isDefined)
+    : currentConfig[CLIENT_CONFIG]
```
Comment (netama) [inline packages/netsuite-adapter/src/config.ts:159]: nit: I'm not sure I'm following the difference between the two cases correctly - is it intended to prioritize `suggestions[FETCH_ALL_TYPES_AT_ONCE]` over `currentConfig[CLIENT_CONFIG]?.[FETCH_ALL_TYPES_AT_ONCE]`? if so - maybe use 
```suggestion
  const clientConfigSuggestion = suggestions[FETCH_ALL_TYPES_AT_ONCE] !== undefined
    ? _.pickBy({
      ...(currentConfig[CLIENT_CONFIG] ?? {}),
      [FETCH_ALL_TYPES_AT_ONCE]: suggestions[FETCH_ALL_TYPES_AT_ONCE],
    }, values.isDefined)
    : currentConfig[CLIENT_CONFIG]
```
or
```suggestion
  const clientConfigSuggestion = _.pickBy(_.merge(
      {},
      currentConfig[CLIENT_CONFIG],
      { [FETCH_ALL_TYPES_AT_ONCE]: suggestions[FETCH_ALL_TYPES_AT_ONCE] },
    ), values.isDefined)
```
to make sure if more parameters are added they won't be missed?

**Review by netama — APPROVED** [review]

File: packages/netsuite-adapter/src/config.ts (hunk 1 above)
Comment (omrilit) [reply packages/netsuite-adapter/src/config.ts]: Great idea!
Implemented it also for the adapter-config level.

**Pull request merged**